public/js/components/SourceFooter.js

```diff
@@ -80,15 +84,44 @@ const SourceFooter = React.createClass({
       "Prettify Source"
     );
   },
+  onKeyUp(e) {
+    const query = e.target.value;
+    const ed = this.props.editor;
+    const ctx = { ed, cm: ed.codeMirror };
+
+    if (e.key != "Enter") {
+      find(ctx, query);
+    } else if (e.shiftKey) {
+      findPrev(ctx, query);
+    } else {
+      findNext(ctx, query);
+    }
+  },
+
+  focusSearch(shortcut, e) {
+    e.stopPropagation();
+    e.preventDefault();
+    const node = findDOMNode(this).querySelector(".source-search");
+    node.focus();
+  },
+
+  setupKeyboardShortcuts() {
```

Comment:
this is a pattern borrowed from the right sidebar. It's designed to be called once post mount